promql/engine.go

```diff
@@ -388,33 +393,49 @@ func (ng *Engine) newQuery(q storage.Queryable, expr parser.Expr, start, end tim
 }
 
 var ErrValidationAtModifierDisabled = errors.New("@ modifier is disabled")
+var ErrValidationNegativeOffsetDisabled = errors.New("negative offset is disabled")
 
 func (ng *Engine) validateOpts(expr parser.Expr) error {
-	if ng.enableAtModifier {
+	if ng.enableAtModifier && ng.enableNegativeOffset {
 		return nil
 	}
 
 	var validationErr error
 	parser.Inspect(expr, func(node parser.Node, path []parser.Node) error {
 		switch n := node.(type) {
 		case *parser.VectorSelector:
-			if n.Timestamp != nil || n.StartOrEnd == parser.START || n.StartOrEnd == parser.END {
+			if !ng.enableAtModifier &&
```

Comment:
This is becoming a bit unreadable.
Can we go for a pattern:
```
var atModifierUsed, negativeOffsetUsed
```

and at the end

```
if  atModifierUsed && !ng.enableAtModified {
 return err
}
```

Reply:
Yes, done.